src/Swarrot/Consumer.php

```diff
@@ -75,25 +76,54 @@ public function consume(array $options = array())
             $this->processor->initialize($options);
         }
 
+        if ($options['method'] === 'consume') {
+            $this->methodConsume($options);
+        } else {
+            $this->methodGet($options);
+        }
+
+
+        if ($this->processor instanceof TerminableInterface) {
+            $this->processor->terminate($options);
+        }
+    }
+
+    private function methodConsume(array $options = array())
+    {
+        $consumerTag = uniqid();
+        try {
+
+            $this->messageProvider->consume($consumerTag, function($message) use ($options) {
+
+                if (false === $this->processor->process($message, $options)) {
+                    throw new \Exception('Stop processing');
+                }
+
+                if ($this->processor instanceof SleepyInterface) {
+                    if (false === $this->processor->sleep($options)) {
+                        throw new \Exception('Stop processing');
+                    }
+                }
+            });
+
+        } catch (\Exception $e) {}
```

Comment:
swallowing any exception silently looks very bad. If the processor throws, this will get lost and debugging will not be possible (you probably won't even get an error report, unless your monitoring considers that a consumer exiting with a success exit code is also an issue, but you will still not know what happened).